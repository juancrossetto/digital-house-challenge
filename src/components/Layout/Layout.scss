
.layout {
  max-width: 80em;
  align-self: center;
  margin: 0 auto;
}

.menu {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;

  &__image-container {
    background-color: #ffffff;
    border-radius: 19px;
    margin-top: 15px;
  }

  &__img {
    width: 7rem;
    padding: 0.5rem;
    &:hover {
      cursor: pointer;
    }
  }

  &__btn {
    border: 1px solid grey;
    padding: 0.3rem;
    border-radius: 0.6rem;
    margin-top: 0.3rem;
    margin-right: 0.3rem;
  }

  &__nav {
    display: flex;
    justify-content: flex-end;
    width: 80%;

    &__item {
      list-style-type: none;
      margin: 0;
      padding: 0;
      margin-right: 3rem;
    }

    &__item-link {
      margin: 0 1rem;
      &:hover {
        cursor: pointer;
        font-weight: bold;
        color: #f52f41;
      }
    }
  }
}


