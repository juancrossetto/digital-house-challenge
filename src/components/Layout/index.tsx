import { Icon, useColorMode } from "@chakra-ui/react";
import React, { FC } from "react";
import { Link, useHistory } from "react-router-dom";
import { BsMoon, BsSun } from "react-icons/bs";
import "./Layout.scss";

import LogoIcon from "../../assets/images/logo.png";

const Layout: FC = ({ children }) => (
  <div className="layout">
    <div>
      <div>
        <Menu />
      </div>
      {children}
    </div>
  </div>
);

const Menu = () => {
  const { push } = useHistory();
  const { colorMode, toggleColorMode } = useColorMode();
  return (
    <div className="menu">
      <div className="menu__image-container">
        <img
          className="menu__img"
          src={LogoIcon}
          alt="logo digital house"
          onClick={() => push({ pathname: "/" })}
          loading="lazy"
          decoding="async"
        />
      </div>
      <ul className="menu__nav">
        <li className="menu__nav__item">
          <Link to="/config" className="menu__nav__item-link">Administrar</Link>
          <Link to="/history" className="menu__nav__item-link">Historial</Link>
        </li>
      </ul>
      <button className="menu__btn" onClick={toggleColorMode}>
        {colorMode === "light" ? <Icon as={BsSun} /> : <Icon as={BsMoon} />}
        {colorMode === "light" ? " Light" : " Dark"} mode
      </button>
    </div>
  );
};

export default Layout;
