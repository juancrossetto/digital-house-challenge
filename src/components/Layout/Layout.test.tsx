import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import Layout from ".";

it("<Layout /> renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Layout>Layout Test</Layout>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
