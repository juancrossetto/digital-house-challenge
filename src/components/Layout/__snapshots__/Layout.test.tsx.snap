// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly 1`] = `
<div
  className="layout"
>
  <div>
    <div>
      <div
        className="menu"
      >
        <div
          className="menu__image-container"
        >
          <img
            alt="logo digital house"
            className="menu__img"
            decoding="async"
            loading="lazy"
            onClick={[Function]}
            src="logo.png"
          />
        </div>
        <ul
          className="menu__nav"
        >
          <li
            className="menu__nav__item"
          >
            <a
              href="/config"
              onClick={[Function]}
            >
              Administrar
            </a>
          </li>
        </ul>
        <button
          className="menu__btn"
        >
          <svg
            className="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M14.53 10.53a7 7 0 01-9.058-9.058A7.003 7.003 0 008 15a7.002 7.002 0 006.53-4.47z"
              fillRule="evenodd"
            />
          </svg>
           Dark
           mode
        </button>
      </div>
    </div>
    Layout Test
  </div>
</div>
`;
